src/contracts/core/AVSDirectory.sol

```diff
@@ -94,16 +146,20 @@ contract AVSDirectory is
             operatorSignature.expiry >= block.timestamp,
             "AVSDirectory.registerOperatorToOperatorSets: operator signature expired"
         );
-        // Assert operator's signature `salt` has not already been spent.
-        require(
-            !operatorSaltIsSpent[operator][operatorSignature.salt],
-            "AVSDirectory.registerOperatorToOperatorSets: salt already spent"
-        );
         // Assert `operator` is actually an operator.
         require(
             delegation.isOperator(operator),
             "AVSDirectory.registerOperatorToOperatorSets: operator not registered to EigenLayer yet"
         );
+        // Assert that the AVS is an operator set AVS.
+        require(
+            isOperatorSetAVS[msg.sender], "AVSDirectory.registerOperatorToOperatorSets: AVS is not an operator set AVS"
```

Comment:
instead, check that the operator set exists here

Reply:
or actually, check in the internal function

Reply:
Operator set existence is checked in the internal function so we don't duplicate code in migration + registration. We check whether AVS is an operator set AVS in the parent function since we don't need to check it for every operatorSetID of the AVS - gas savings, even though its a warm sload